docs/library/userguide.md

```diff
@@ -76,13 +75,36 @@ if not verification_output:
 
 ```
 
-### Rendering QREF files using `qref-render` (experimental)
+### Topology validation
+
+There can be cases where a program is correct from the perspective of Pydantic validation, but has incorrect topology. This includes cases such as:
+
+- Disconnected ports
+- Ports with multiple connections
+- Cycles in the graph
 
-!!! Warning
+In order to validate whether the topology of the program is correct you can use `verify_topology` method. Here's a short snippet showing how one can verify their program and print out the problems (if any).
+
+```python
+from qref.verification import verify_topology
 
-    This feature is considered experimental and may occassionally produce
-    incorrect results.
+program = load_some_program()
 
+verification_output = verify_topology(program)
+
+if not verification_output:
+    print("Program topology is incorrect, due to the following issues:")
+    for problem in verification_output.problems:
+        print(problem)
+
+```
+
+### Rendering QREF files using `qref-render` (experimental)
+
+<div style="padding: 15px; border: 1px solid transparent; border-color: transparent; margin-bottom: 20px; border-radius: 4px; color: #8a6d3b;; background-color: #fcf8e3; border-color: #faebcc;">
```

Comment:
[question] Why did you change it from `!!!Warning` to this elaborate html?

Reply:
I didn't see the alert highlight in the preview, but only the text `!!!Warning,` so I thought it might be a mistake; If it's intentional, I will leave it unchanged.

Reply:
Did you build the docs (i.e. `mkdocs serve`) or just used preview in the editor you're using?
Cause it renders properly with the `mkdocs serve` for me.

Reply:
Solved in commit 1755e67.